### ISSUE TITLE
Expand fuzzer coverage

### ISSUE THREAD

**Issue by timwoj** [zeek/zeek#2142]
We currently have coverage for raw packets, pop3, and dns (in a fashion). It would be good to expand our coverage to other major protocols. I'm currently thinking at least the following:

- DNS again (using the buffer splitter code in the pop3 fuzzer)
- HTTP
- ICMP
- SMTP

On top of that, we should take a look at the coverage statistics we're getting from oss-fuzz and determine where the gaps are in the current fuzzers and corpus.

**Comment by timwoj** [comment]
The best way to create corpora may be to pick out a few representative traces for each protocol from our traces library (or from other places) and just run them through https://github.com/JustinAzoff/pcap_simplify.

**Comment by timwoj** [comment]
Another good thing to add here would be fuzzers for packet analyzers.

**Comment by timwoj** [comment]
While looking at the coverage numbers of oss-fuzz, @JustinAzoff discovered that we're potentially missing a ton of coverage because the event handlers aren't defined and so checking for them always returns false. I did a bit of testing with an SMTP analyzer that I added and I can gain about 2% coverage with just forcing `EventHandler::operator bool()` to always return true.

We have a BIF method in script land called `generate_all_events`, but we can't just use the code from there because the event registry is never setup as part of the fuzzing runs. We'll need to add a global variable or something that can be set at the start of fuzzing and checked for by that operator.
